packages/tinacms-authjs/src/index.ts

```diff
@@ -133,10 +134,61 @@ const TinaCredentialsProvider = ({
       authenticate(databaseClient, credentials.username, credentials.password),
   })
 
+const AuthJsBackendAuthentication = ({
+  authOptions,
+}: {
+  authOptions: AuthOptions
+}) => {
+  const nextBackendAuthentication: BackendAuthentication = {
+    isAuthorized: async (req, res) => {
+      // @ts-ignore
+      const session = await getServerSession(req, res, authOptions)
+
+      // @ts-ignore
+      if (!req.session) {
+        Object.defineProperty(req, 'session', {
+          value: session,
+          writable: false,
+        })
+      }
+
+      if (!session?.user) {
+        return {
+          errorCode: 401,
+          errorMessage: 'Unauthorized',
+          isAuthorized: false,
+        }
+      }
+      if ((session?.user as any).role !== 'user') {
+        return {
+          errorCode: 403,
+          errorMessage: 'Forbidden',
+          isAuthorized: false,
+        }
+      }
+      return { isAuthorized: true }
+    },
+    extraRoutes: {
+      auth: {
+        isAuthRequired: false,
+        handler: async (req, res) => {
+          // @ts-ignore
+          const { routes } = req.query
+          const [, ...rest] = routes
+          // @ts-ignore
+          req.query.nextauth = rest
+          NextAuth(authOptions)(req, res)
+        },
+      },
+    },
+  }
+  return nextBackendAuthentication
+}
 export {
   TinaCredentialsProvider,
   TinaAuthJSOptions,
   withAuthJSApiRoute,
+  AuthJsBackendAuthentication,
   // TODO: This probably needs a better name?
   createAuthJSApiRoute,
```

Comment:
```suggestion
```

Reply:
also remove `withAuthJSApiRoute` and `createAuthJsApiRoute`

Reply:
removed